src/cryptography/utils.py

```diff
@@ -30,6 +30,10 @@ def _check_bytes(name, value):
         raise TypeError("{0} must be bytes".format(name))
 
 
+def _check_byteslike(name, value):
+    memoryview(value)
```

Comment:
Hmm, so this results in a worse error message than previously, do we care?

Reply:
I think I'm okay with it. That said, we could do:

```python
try:
    memoryview(value)
except TypeError:
    raise TypeError("{0} must be bytes-like.".format(name))
```